### ISSUE TITLE
Unable to pass string array literal to contains() interpolation function

### ISSUE THREAD

**Issue by mpkerr** [hashicorp/terraform#16914]
### Terraform Version

```
terraform -v
Terraform v0.10.7
```

### Terraform Configuration Files

```hcl
resource "aws_route53_record" "zeus_elb_alias" {
  count   = "${contains( ["prod", "sbx", "perf"], var.planet) ? 1 : 0}"
  zone_id = "${var.external_dns_zone}"
  name    = "${data.aws_route53_zone.external_dns_zone.name}"
  type    = "A"

  alias {
    name                   = "${var.routing_service_address}"
    zone_id                = "${aws_elb.zeuselb.zone_id}"
    evaluate_target_health = false
  }
}
```

### Expected Behavior

`terraform fmt` does not complain about this construct. however `terraform get` results in the below error. The problem is that it does not like the literal array of strings in the contains() interpolation function. If I move that list to a locals block local variable and reference the variable instead this code works.

### Actual Behavior

```sh
mkerr-macOS:zeus mkerr$ terraform get
Error loading configuration: Error loading /Users/mkerr/Projects/tools/terraform-modules/aws/zeus/zeus.tf: parse error at 1:12: expected expression but found "["
```

**Comment by apparentlymart** [comment]
Hi @mpkerr! Sorry for this confusing error.

The current interpolation language does not support list literals using `[` and `]` delimiters. Instead, it uses a list constructor function called `list`:

```hcl
  count = "${contains(list("prod", "sbx", "perf"), var.planet) ? 1 : 0}"
```

We are currently working on integrating a revamped language parser/evaluator that merges together the structural language (HCL) with the interpolation language (HIL) so that we can support the following, more intuitive usage where both are handled at once as a single language:

```hcl
  # NOT YET IMPLEMENTED: details may change before release
  count = contains(["prod", "sbx", "perf"], var.planet) ? 1 : 0
```

An initial version of this new parser should be available as an opt-in alternative in the near future to gather feedback and catch bugs, and then in a future major release the new parser will be used by default to allow expressions like the above.

**Comment by apparentlymart** [comment]
Hi again @mpkerr!

I'm pleased to report that the changes I mentioned previously are now merged into master and will be included in the forthcoming v0.12.0 release. I verified this with `terraform console` in the v0.12.0-alpha1 prerelease build:

```
$ terraform console
> "${contains(["prod", "sbx", "perf"], "perf") ? 1 : 0}"
1
> "${contains(["prod", "sbx", "perf"], "prof") ? 1 : 0}"
0
> contains(["prod", "sbx", "perf"], "perf")
true
```
